crates/egui/src/memory/mod.rs

```diff
@@ -1083,19 +1116,34 @@ impl Memory {
     }
 
     /// Open the given popup and close all others.
+    ///
+    /// Note that you must call `keep_popup_open` on subsequent frames as long as the popup is open.
     pub fn open_popup(&mut self, popup_id: Id) {
-        self.popup = Some((popup_id, None));
+        self.popup = Some(OpenPopup::new(popup_id, None));
+    }
+
+    /// Popups must call this every frame while open.
+    ///
+    /// This is needed because in some cases popups can go away without `close_popup` being
+    /// called. For example, when a context menu is open and the underlying widget stops
+    /// being rendered.
+    pub fn keep_popup_open(&mut self, popup_id: Id) {
```

Comment:
I feel like there is a more elegant solution to this problem that I'm not seeing 😆 

We could replace `is_popup_open` with `show_popup_if_open(popup_id, closure)`, but that's a bit annoying too (especially when it comes to lifetimes and locks), so… no.

I'm also not in love with this function name, but again fail to come up with something better.

so… LGTM :)